internal/services/mssqlmanagedinstance/mssql_managed_instance_start_stop_schedule_resource.go

```diff
@@ -0,0 +1,330 @@
+				properties.Properties.Description = pointer.To(model.Description)
+			}
+
+			if metadata.ResourceData.HasChange("schedule") {
+				properties.Properties.ScheduleList = expandScheduleItemModelArray(model.Schedule)
+			}
+
+			if metadata.ResourceData.HasChange("timezone_id") {
+				properties.Properties.TimeZoneId = pointer.To(model.TimeZoneId)
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, managedInstanceID, *properties); err != nil {
+				return fmt.Errorf("updating %s: %+v", managedInstanceID, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MSSQLManagedInstance.ManagedInstanceStartStopSchedulesClient
+
+			id, err := parse.ManagedInstanceStartStopScheduleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			managedInstanceID := commonids.NewSqlManagedInstanceID(id.SubscriptionId, id.ResourceGroup, id.ManagedInstanceName)
+
+			resp, err := client.Get(ctx, managedInstanceID)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(managedInstanceID)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", managedInstanceID, err)
+			}
+
+			state := SqlManagedInstanceStartStopScheduleModel{
+				SqlManagedInstanceId: managedInstanceID.ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				if properties := model.Properties; properties != nil {
+					state.Description = pointer.From(properties.Description)
+
+					state.NextExecutionTime = pointer.From(properties.NextExecutionTime)
+
+					state.NextRunAction = pointer.From(properties.NextRunAction)
+
+					if properties.ScheduleList != nil {
+						state.Schedule = flattenScheduleItemModelArray(properties.ScheduleList)
+					}
+
+					state.TimeZoneId = pointer.From(properties.TimeZoneId)
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r MsSqlManagedInstanceStartStopScheduleResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MSSQLManagedInstance.ManagedInstanceStartStopSchedulesClient
+
+			id, err := parse.ManagedInstanceStartStopScheduleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			managedInstanceID := commonids.NewSqlManagedInstanceID(id.SubscriptionId, id.ResourceGroup, id.ManagedInstanceName)
+
+			if _, err := client.Delete(ctx, managedInstanceID); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandScheduleItemModelArray(inputList []ScheduleItemModel) []schedule.ScheduleItem {
+	outputList := make([]schedule.ScheduleItem, 0, len(inputList))
+
+	for _, v := range inputList {
+		input := v
```

Comment:
Assigning `v` to `input` isn't necessary, we can just rename `v` in the for loop declaration to `input` if that's the var name we want

```suggestion
	for _, input := range inputList {
```